src/App.tsx

```diff
@@ -1,9 +1,24 @@
-import React from 'react';
+import React, { useState, ChangeEvent } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const visibleMovies = moviesFromServer.filter(movie => {
+    const preparedQuery = query.toLowerCase().trim();
+    const preparedTitle = movie.title.toLowerCase();
+    const preparedDesc = movie.description.toLowerCase();
+
+    return preparedTitle.includes(preparedQuery)
+        || preparedDesc.includes(preparedQuery);
+  });
+
+  const changeSearchInput = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
good practice to name the event handler as handle[eventName]
```suggestion
  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
```